kernel/src/client/default/parquet.rs

```diff
@@ -50,14 +52,32 @@ impl<E: TaskExecutor> ParquetHandler for DefaultParquetHandler<E> {
         physical_schema: SchemaRef,
         _predicate: Option<Expression>,
     ) -> DeltaResult<FileDataReadResultIterator> {
-        // TODO at the very least load only required columns ...
         if files.is_empty() {
             return Ok(Box::new(std::iter::empty()));
         }
 
         let arrow_schema: ArrowSchemaRef = Arc::new(physical_schema.as_ref().try_into()?);
-        let file_reader = ParquetOpener::new(1024, physical_schema.clone(), self.store.clone());
-        let mut stream = FileStream::new(files.to_vec(), arrow_schema, file_reader)?;
+        // get the first FileMeta to decide how to fetch the file
+        // s3://    -> aws   (ParquetOpener)
+        // nothing  -> local (ParquetOpener)
+        // https:// -> assume presigned URL (and fetch without object_store)
+        //   -> reqwest to get data
+        //   -> parse to parquet
+        // SAFETY: we did is_empty check above, this is ok.
+        let mut stream: std::pin::Pin<
+            Box<dyn futures::Stream<Item = DeltaResult<arrow_array::RecordBatch>> + Send>,
+        > = match files[0].location.scheme() {
+            "http" | "https" => Box::pin(FileStream::new(
+                files.to_vec(),
+                arrow_schema,
+                PresignedUrlOpener::new(1024, physical_schema.clone()),
+            )?),
+            _ => Box::pin(FileStream::new(
```

Comment:
same comment as above, probably want to actually parse `s3://` etc. then `_` for error/unsupported?

Reply:
same as above, I think it's actually better to let the object_store just tell us it can't handle a particular url